### ISSUE TITLE
Warning for unnecessary qualifications of names?

### ISSUE THREAD

**Issue by pikajude** [rust-lang/rust-clippy#4707]
Hi, sorry if this is already covered by clippy but I couldn't find it in the lints.

``` rust
use std::io::Result;

fn dosomething() -> std::io::Result {
  println!("Hello, world!")
}
```

Is it possible to make clippy warn on the usage of `std::io::Result` on line 3, since the unqualified name `Result` is in scope and doesn't collide with any other imported name?

**Comment by jplatte** [comment]
This is a rustc lint now, no need to add it to clippy: https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unused-qualifications

**Comment by giraffate** [comment]
Thanks! I'm closing this because of https://github.com/rust-lang/rust-clippy/issues/4707#issuecomment-999700713.

**Comment by mattfbacon** [comment]
So rustc lints don't have configuration but I would like to be able to configure `unused_qualifications` to allow `self` and `crate` ... maybe we do need it in clippy?
